internal/icinga2/client_api.go

```diff
@@ -322,10 +322,12 @@ func (client *Client) checkMissedChanges(ctx context.Context, objType string, ca
 		case catchupEventCh <- &catchupEventMsg{eventMsg: &eventMsg{ev, attrs.LastStateChange.Time()}}:
 			stateChangeEvents++
 			if fakeEv.Type == event.TypeAcknowledgementSet {
+				ackEvent := *fakeEv
+				ackEvent.ID = 0 // We don't want to reference a previously sent event, so reset its ID!
```

Comment:
> duplicate the whole event before sending it to any channel

By that, I meant before `fakeEv` is sent to any channel for the first time, i.e. before line 305. That way, the pointer wasn't yet passed anywhere else and duplicating it is safe for sure. So that would mean creating a second variable next to `fakeEv` and duplicating the event right after `client.buildAcknowledgementEvent()`.